File: 1a/LU with pivot/swapRowsV2.py

# we write a more efficient function that swaps rows u and v of some matrix A
def swapRowsV2(A, u, v):

	# we directly swap rows u and v of A without copying
	for i, val in enumerate(A[u]):
		A[u][i] = A[v][i]
		A[v][i] = val

File: 1a/LU with pivot/swapRowsV2.jl

# we write a more efficient function that swaps rows u and v of some matrix A
function swapRowsV2!(A, u, v)

	# we directly swap rows u and v of A without copying
	temp = A[u, :]
	A[u, :] = A[v, :]
	A[v, :] = temp
end
